plugins/modules/vmware_guest.py

```diff
@@ -3091,10 +3091,11 @@ def deploy_vm(self):
         clonespec = None
         clone_method = None
         try:
+            # Only select specific host when ESXi hostname is provided
+            if self.params['esxi_hostname']:
+                self.relospec.host = self.select_host()
```

Comment:
I didn't test this, but it looks like the else branch doesn't use `self.relospec`. I think you need to define the hostname to deploy to in `self.relospec` in the if branch, and in `self.configspec` in the else branch.

As I've said, I didn't test this. I'm just trying to understand what the code does and maybe I'm wrong.